rpi3-convert-stage-5.sh

```diff
@@ -16,26 +16,60 @@
 
 application_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 output_dir=${application_dir}/output
+uboot_dir=${output_dir}/uboot-mender
 bin_dir_pi=${output_dir}/bin/raspberrypi
 sdimg_base_dir=$output_dir/sdimg
 
 echo "Running: $(basename $0)"
 declare -a sdimgmappings
 declare -a sdimg_partitions=("boot" "primary" "secondary" "data")
 
+# Takes following arguments:
+#
+#  $1 - ARM toolchain
 build_uboot_files() {
+  local CROSS_COMPILE=${1}-
+  local ARCH=arm
+  local revision="mender-rpi-2017.09"
+  local uboot_repo_vc_dir=$uboot_dir/.git
+  local repo_clean=0
+
+  export CROSS_COMPILE=$CROSS_COMPILE
+  export ARCH=$ARCH
+
   mkdir -p $bin_dir_pi
 
-  echo -e "Downloading U-Boot related files..."
+  echo -e "Building U-Boot related files..."
+
+  if [ ! -d $uboot_repo_vc_dir ]; then
+    git clone https://github.com/mendersoftware/uboot-mender.git -b $revision
+    repo_clean=1
+  fi
+
+  cd $uboot_dir
+
+  [[ repo_clean -eq 0 ]] && { make --quiet clean; make --quiet distclean; }
+
+  make rpi_3_32b_defconfig && make
+  make envtools
+
+  cat<<-'EOF' >boot.cmd
+	fdt addr ${fdt_addr} && fdt get value bootargs /chosen bootargs
+	run mender_setup
+	mmc dev ${mender_uboot_dev}
+	load ${mender_uboot_root} ${kernel_addr_r} /boot/zImage
+	bootz ${kernel_addr_r} - ${fdt_addr}
```

Comment:
Dont you need to escape `$` here? 

It seems that you do not after I tested running the command. I am curios do you know why this is?

Reply:
Expansion is disabled by quoting the EOF-doc delimiter.

Reply:
Cool, learnt something new :)